### ISSUE TITLE
feat(embedding): FastText domain adaptation training — standalone CLI tool

### ISSUE THREAD

**Issue by petitan** [petitan/IronBase#45]
## Kontextus

A Tudástár chatbot `rag_search`-e nem találja meg a "PEF-35 fékpad" keresésre a helyes dokumentumot, mert a "PEF-35" termékkód OOV (Out of Vocabulary) a FastText modellben. A cosine similarity "PEF-35" ↔ "fékpad" között mindössze **0.088** — gyakorlatilag random.

**Megoldás:** A FastText modellt a domain-specifikus corpuson (árajánlatok, kezelési útmutatók) fine-tune-olni, hogy a termékkódok (PEF-35, M1N1, SICE) szemantikailag közel kerüljenek a kapcsolódó szavakhoz.

## Döntés: standalone CLI tool, NEM MCP tool

A training egy egyszeri/ritka művelet (~5 perc runtime, ~300 MB RAM). **Nincs keresnivalója az MCP szerveren belül.** Nem módosítja az IronBase kódot, nem bővíti az API-t, nem ad hozzá új JobType-ot.

**Precedens:** `ironbase-backup` — már létező standalone crate a workspace-ben.

### Új crate: `ironbase-training`

```
MongoLite/
├── ironbase-training/        # ÚJ standalone crate
│   ├── Cargo.toml
│   └── src/
│       ├── main.rs           # CLI (clap)
│       ├── corpus.rs         # Corpus építés IronBase collection-ből
│       ├── trainer.rs        # Skipgram + negative sampling
│       └── ibv2_writer.rs    # IBv2 formátum output
```

### CLI interface

```bash
ironbase-training \
  --db /path/to/ironbase_data.mlite \
  --collection documents \
  --text-field content_text \
  --base-model /path/to/cc.hu.300.ironbase.v2.bin \
  --output /path/to/cc.hu.300.finetuned.v2.bin \
  --epochs 5 \
  --min-count 2 \
  --window-size 5 \
  --neg-samples 5 \
  --alpha 0.025 \
  --min-alpha 0.0001
```

### Használat utána

```bash
# 1. Training
ironbase-training --db ... --collection ... --output finetuned.v2.bin

# 2. Env var átállítás
export IRONBASE_FASTTEXT_MODEL=/path/to/finetuned.v2.bin

# 3. MCP server restart (automatikusan detektálja a modellváltást, re-embed)
```

## Training algoritmus

**Skipgram + negative sampling** (FastText paper alapján):

1. **Corpus építés:** IronBase collection → dokumentumok text_field-jéből tokenizált mondatok
2. **Vocab építés:** szó → gyakoriság, szűrés `min_count`-ra
3. **Input vektorok:** base modellből inicializálva (mmap, `word_vector` vagy `compute_subword_vector`)
4. **Output vektorok:** zero-init, csak training vocab méretű
5. **Negative sampling tábla:** frekvencia^0.75 eloszlás
6. **Training loop:**
   ```
   for epoch in 0..epochs:
     for sentence in shuffled_corpus:
       for (i, target) in sentence:
         lr = linear_decay(alpha → min_alpha, progress)
         window = rand(1..=window_size)
         for context in sentence[i-window..i+window]:
           // Pozitív: sigmoid(dot(input, output[context])) → gradient
           // K negatív: random szavak → gradient
           // Update: input_vec, output_vec, bucket_deltas
   ```
7. **IBv2 output:** eredeti mmap szekvenciális olvasása, módosított vektorok behelyettesítése, bucket szekció: eredeti + delta

### Memória stratégia

| Komponens | Méret |
|-----------|-------|
| Base modell (read-only mmap) | ~0 MB RSS (OS page cache) |
| Corpus tokenizálva | ~25 MB |
| Input vektorok (50K × 300 × f32) | ~60 MB |
| Output vektorok (50K × 300 × f32) | ~60 MB |
| Bucket delták (100K × 300 × f32) | ~120 MB |
| Neg. sampling tábla (10M × u32) | ~40 MB |
| **Összesen** | **~307 MB** |

## Implementációs szabályok

A CLAUDE.md Kód Konzisztencia Protokollt követve:

- **Mintának használandó:** `ironbase-backup/` crate struktúra (CLI, Cargo.toml)
- **FastText formátum:** `mcp-server/src/embedding/fasttext.rs` IBv2 reader logikája (header parsing, mmap, word_index, subword buckets)
- **Error handling:** `thiserror` (mint mindenhol a workspace-ben)
- **CLI:** `clap` derive (mint `mcp-server`, `ironbase-cli`)
- **Dependency:** `ironbase-core` (collection olvasáshoz), `memmap2` (base modell)

## Verifikáció

1. `cargo build -p ironbase-training` — hiba nélkül
2. `cargo test -p ironbase-training` — unit tesztek (vocab, neg table, sigmoid, xorshift, IBv2 roundtrip)
3. Manuális teszt:
   - `ironbase-training --db ... --collection documents --output finetuned.v2.bin`
   - `IRONBASE_FASTTEXT_MODEL=finetuned.v2.bin` → szerver restart
   - `embed_text "PEF-35"` vs `embed_text "fékpad"` → cosine similarity >> 0.088
   - Tudástár: "PEF-35 fékpad" → PEF-35 doc top-1

## Ami NEM változik

- ❌ MCP server kód (nincs új tool, nincs dispatcher módosítás)
- ❌ `FastTextProvider` API (nincs pub accessor bővítés)
- ❌ `JobType` enum (nincs `EmbedFinetune` variant)
- ❌ `tools/definitions/` (nincs új tool séma)
- ❌ MCP server Cargo.toml (nincs verzió bump)
